airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -41,8 +45,10 @@ class GoogleCloudBaseHook(BaseHook, LoggingMixin):
     All hook derived from this base hook use the 'Google Cloud Platform' connection
     type. Two ways of authentication are supported:
 
-    Default credentials: Only specify 'Project Id'. Then you need to have executed
-    ``gcloud auth`` on the Airflow worker machine.
+    Default credentials: Only the 'Project Id' is required. You'll need to
+    have set up default credentials, such as by the
+    ``GOOGLE_APPLICATION_DEFAULT`` environment variable or from the metadata
```

Comment:
Is this going to work with users that don't want to use a service account? Developers often times want to oauth using their Google account rather than using a service account. Will this continue to work?

Reply:
Yes, it will work. `google-auth` does read credentials from the Google Cloud SDK.

Note: those developers will get an annoying warning message. See: https://github.com/GoogleCloudPlatform/google-auth-library-python/issues/271

If that's a concern we could possibly add a user-auth option as is done in pandas-gbq, but that's quite a bit more code to do. I'd prefer to do that in a separate PR, as it would be additional functionality as opposed to this which is a 1:1 replacement.

Reply:
Great, I'm fine with a 1:1 replacement. Just want to make sure that we don't lose functionality that we rely on. :)